pkg/controller/master/node/promote_controller.go

```diff
@@ -246,53 +247,63 @@ func (h *PromoteHandler) setPromoteResult(job *batchv1.Job, node *corev1.Node, s
 // If the cluster doesn't need to be promoted, return nil
 func selectPromoteNode(nodeList []*corev1.Node) *corev1.Node {
 	var (
-		managementNumber     int
-		managementRoleNumber int
```

Comment:
we also need to add two cases:
```go
   wu1 = NewDefaultNodeBuilder().Name("w-unknown-1").Harvester().Unknown().Worker()
	wc1 = NewDefaultNodeBuilder().Name("w-complete-1").Harvester().Complete().Worker()

{
			name: "one management and one promote unknown worker and one worker",
			args: args{
				nodeList: []*corev1.Node{m1, wu1, w1},
			},
			want: nil,
		},
		{
			name: "one management and one promoted worker and one worker",
			args: args{
				nodeList: []*corev1.Node{m1, wc1, w1},
			},
			want: nil,
		},
```
and these cases will failed after remove variable `managementRoleNumber` and turn method `hasPromoteStatus` to unused

Reply:
```go
func (n *NodeBuilder) Unknown() *NodeBuilder {
	n.node.Annotations[HarvesterPromoteStatusAnnotationKey] = PromoteStatusUnknown
	return n
}
```

Reply:
Yes they are covered now in the tests (including the zone aware ones) and also in the selectPromoteNode method.